pkg/maps/act/act.go

```diff
@@ -0,0 +1,169 @@
+		return
+	}
+
+	out.NodeDefines = map[string]string{
+		"ENABLE_ACTIVE_CONNECTION_TRACKING": "1",
+		"LB_ACT_MAP":                        ActiveConnectionTrackingMapName,
+	}
+
+	out.MapOut = bpf.NewMapOut(ActiveConnectionTrackingMap(createActiveConnectionTrackingMap(in.Lifecycle)))
+	return
+}
+
+func createActiveConnectionTrackingMap(lc cell.Lifecycle) *actMap {
+	m := bpf.NewMap(ActiveConnectionTrackingMapName,
+		ebpf.LRUHash,
+		&ActiveConnectionTrackerKey{},
+		&ActiveConnectionTrackerValue{},
+		65536,
+		0,
+	)
+
+	lc.Append(cell.Hook{
+		OnStart: func(context cell.HookContext) error {
+			return m.OpenOrCreate()
+		},
+		OnStop: func(context cell.HookContext) error {
+			return m.Close()
+		},
+	})
+
+	return &actMap{m}
+}
+
+func (m actMap) IterateWithCallback(cb ActiveConnectionTrackingIterateCallback) error {
+	return m.m.DumpWithCallback(func(k bpf.MapKey, v bpf.MapValue) {
+		key := k.(*ActiveConnectionTrackerKey)
+		value := v.(*ActiveConnectionTrackerValue)
+
+		cb(key, value)
+	})
+}
+
+// ActiveConnectionTrackerKey is the key to ActiveConnectionTrackingMap.
+//
+// It must match 'struct lb_act_key' in "bpf/lib/common.h".
+// +k8s:deepcopy-gen=true
+// +k8s:deepcopy-gen:interfaces=github.com/cilium/cilium/pkg/bpf.MapKey
+type ActiveConnectionTrackerKey struct {
+	SvcID uint16 `align:"svc_id"`
+	Zone  uint8  `align:"zone"`
+	Pad   uint8  `align:"pad"`
+}
+
+func (s *ActiveConnectionTrackerKey) New() bpf.MapKey { return &ActiveConnectionTrackerKey{} }
+
+func (k *ActiveConnectionTrackerKey) String() string {
+	v := k.ToHost()
+	if svcAddr, err := service.GetID(uint32(v.SvcID)); err == nil {
+		return fmt.Sprintf("%s[%s]", svcAddr.String(), option.Config.GetZone(v.Zone))
+	}
+	return fmt.Sprintf("%d[%s]", v.SvcID, option.Config.GetZone(v.Zone))
+}
+
+func (k *ActiveConnectionTrackerKey) GetKeyPtr() unsafe.Pointer { return unsafe.Pointer(k) }
+func (k *ActiveConnectionTrackerKey) NewValue() bpf.MapValue    { return &ActiveConnectionTrackerValue{} }
+
+func (k *ActiveConnectionTrackerKey) ToNetwork() *ActiveConnectionTrackerKey {
+	n := *k
+	return &n
+}
+
+func (k *ActiveConnectionTrackerKey) ToHost() *ActiveConnectionTrackerKey {
+	h := *k
+	h.SvcID = byteorder.NetworkToHost16(h.SvcID)
+	return &h
+}
+
+// ActiveConnectionTrackerValue must match 'struct lb_act_value' in "bpf/lib/common.h".
+// +k8s:deepcopy-gen=true
+// +k8s:deepcopy-gen:interfaces=github.com/cilium/cilium/pkg/bpf.MapValue
+type ActiveConnectionTrackerValue struct {
+	Opened uint32 `align:"opened"`
+	Closed uint32 `align:"closed"`
+}
+
+func (s *ActiveConnectionTrackerValue) New() bpf.MapValue { return &ActiveConnectionTrackerValue{} }
+
+func (s *ActiveConnectionTrackerValue) String() string {
+	return fmt.Sprintf("+%d -%d", s.Opened, s.Closed)
+}
+
+func (s *ActiveConnectionTrackerValue) GetValuePtr() unsafe.Pointer { return unsafe.Pointer(s) }
+
+func (s *ActiveConnectionTrackerValue) ToNetwork() *ActiveConnectionTrackerValue {
+	n := *s
```

Comment:
These do not actually convert between endianess correctly. I also don't see these used anywhere, nor are these methods part of the required interface for a bpf.MapValue. Could you remove all methods that are not strictly needed? Same for `ActiveConnectionTrackerKey`.